lnwire/funding_locked.go

```diff
@@ -0,0 +1,139 @@
+package lnwire
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/roasbeef/btcd/btcec"
+	"github.com/roasbeef/btcd/wire"
+)
+
+// FundingLocked is the message that both parties to a new channel creation
+// send once they have observed the funding transaction being confirmed on
+// the blockchain. FundingLocked contains the signatures necessary for the
+// channel participants to advertise the existence of the channel to the
+// rest of the network.
+type FundingLocked struct {
+	// ChannelOutpoint is the outpoint of the channel's funding
+	// transaction. This can be used to query for the channel in the
+	// database.
+	ChannelOutpoint *wire.OutPoint
+
+	// ChannelId serves to uniquely identify the channel created by the
+	// current channel funding workflow.
+	ChannelID ChannelID
+
+	// NextPerCommitmentPoint is the secret that can be used to revoke
+	// the next commitment transaction for the channel.
+	NextPerCommitmentPoint *btcec.PublicKey
+}
+
+// NewFundingLocked creates a new FundingLocked message, populating it with
+// the necessary IDs and revocation secret..
+func NewFundingLocked(op *wire.OutPoint, cid ChannelID, ans, abs *btcec.Signature,
```

Comment:
Signatures are no longer a part of the `FundingLocked` message. As a result this function can be modified to omit the signatures.